### PR TITLE
Add support for loading ASM from RSA install location

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

JIRA: https://jira.autodesk.com/browse/DYN-2047

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang @mjkkirschner

### CODE REVIEW

File: src/Tools/DynamoShapeManager/Utilities.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public static class Utilities
         /// <summary>
         /// Key words for Products containing ASM binaries
         /// </summary>
-        private static readonly List<string> ProductsWithASM = new List<string>() { "Revit", "Civil", "FormIt" };
+        private static readonly List<string> ProductsWithASM = new List<string>() { "Revit", "Civil", "Robot Structural Analysis", "FormIt" };
```
Comment (aparajit-pratap) [inline src/Tools/DynamoShapeManager/Utilities.cs]: ASM from RSA will be used if there is no Revit or Civil3D installed on the user's machine.

File: src/Tools/DynamoInstallDetective/ProductLookUp.cs [hunk 2]
```diff
@@ -159,7 +159,7 @@ static string GetDisplayName(RegistryKey key)
         public InstalledProductLookUp(string lookUpName, string fileLookup)
         {
             this.ProductLookUpName = lookUpName;
-            this.fileLocator = (path) => Directory.GetFiles(path, fileLookup).FirstOrDefault();
+            this.fileLocator = (path) => Directory.GetFiles(path, fileLookup, SearchOption.AllDirectories).FirstOrDefault();
```
Comment (QilongTang) [inline src/Tools/DynamoInstallDetective/ProductLookUp.cs]: @aparajit-pratap Did you notice any noticeable slow down with such searching option with RSA?

File: src/Tools/DynamoShapeManager/Utilities.cs (hunk 1 above)
Comment (QilongTang) [reply src/Tools/DynamoShapeManager/Utilities.cs]: I forgot if changing the order here affect the ASM paths found?

File: src/Tools/DynamoInstallDetective/ProductLookUp.cs (hunk 2 above)
Comment (aparajit-pratap) [reply src/Tools/DynamoInstallDetective/ProductLookUp.cs]: I didn't notice any difference as I don't suspect there is a noticeable slowdown if any after this change. In any case, I think this part of the fix might be unavoidable due to the fact that there is no guarantee that ASM will always be found in the root directory of an installation.

File: src/Tools/DynamoShapeManager/Utilities.cs (hunk 1 above)
Comment (aparajit-pratap) [reply src/Tools/DynamoShapeManager/Utilities.cs]: Yes, it does. See: https://github.com/DynamoDS/Dynamo/blob/7a392ef25a532e0782caf57876aa43f0e4fc7ac2/src/Tools/DynamoShapeManager/Utilities.cs#L154

We first iterate over the different LibG versions and then over the supported products list and whichever matches first, comes out the winner. So it does matter if we play with the order. For example if Revit and RSA are both installed, and there is a match, ASM will be loaded from Revit and if we swap Revit with RSA, ASM will be loaded from RSA.

Comment (QilongTang) [reply src/Tools/DynamoShapeManager/Utilities.cs]: Thanks, this is refreshing to me. Keeping the order of Revit first looks safe.

File: src/Tools/DynamoInstallDetective/ProductLookUp.cs (hunk 2 above)
Comment (QilongTang) [reply src/Tools/DynamoInstallDetective/ProductLookUp.cs]: @aparajit-pratap Thank you, sounds reasonable here.

**Comment by aparajit-pratap** [comment]
Thanks @QilongTang for the review.

**Pull request merged**